server/bleep/src/webserver.rs

```diff
@@ -65,33 +66,45 @@ pub async fn start(app: Application) -> Result<()> {
         .route("/answer", get(answer::handle));
 
     if app.env.allow_github_device_flow() {
-        router = router
+        api = api
             .route("/remotes/github/login", get(github::login))
             .route("/remotes/github/status", get(github::status));
     }
 
     if app.env.allow_path_scan() {
-        router = router.route("/repos/scan", get(repos::scan_local));
+        api = api.route("/repos/scan", get(repos::scan_local));
     }
 
     // Note: all routes above this point must be authenticated.
-    if app.env.use_aaa() {
-        router = aaa::router(router, app.clone());
+    if let Environment::PrivateServer = app.env {
+        api = aaa::router(api, app.clone());
     }
 
-    router = router
+    api = api
         .route("/api-doc/openapi.json", get(openapi_json::handle))
         .route("/api-doc/openapi.yaml", get(openapi_yaml::handle))
         .route("/health", get(health));
 
-    let router: Router<()> = router
+    let api: Router<()> = api
         .layer(Extension(app.indexes.clone()))
         .layer(Extension(app.semantic.clone()))
         .layer(Extension(app.clone()))
-        .with_state(app)
+        .with_state(app.clone())
         .layer(CorsLayer::permissive())
         .layer(CatchPanicLayer::new());
 
+    let mut router = Router::new().nest("/api", api);
+
+    if let Environment::PrivateServer = app.env {
+        router = router.merge(SpaRouter::new(
+            "/",
+            app.config
+                .frontend_dist
+                .as_ref()
+                .expect("need path to built frontend folder"),
```

Comment:
I don't think this should be a hard error, just continue if not found. We could have an API running separately from the frontend assets due to eg. CDN deployments. A hard error makes this implementation dependent instead of configuration dependent.